server/broadcast.go

```diff
@@ -429,6 +486,11 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 		sess = newSess
 	}
 
+	cxn.sessManager.sessLock.Lock()
+	sess.SegsInFlight = append(sess.SegsInFlight, SegFlightMetadata{startTime: time.Now(), segDur: time.Duration(int64(seg.Duration*1e9)) * time.Nanosecond})
```

Comment:
Could use `time.Duration(seg.Duration * float64(time.Second))` since I think seg.Duration is already a denominated in seconds to avoid extra unit conversions.